Composer/packages/client/src/components/AddRemoteSkillModal/CreateSkillModal.tsx

```diff
@@ -180,21 +196,54 @@ export const CreateSkillModal: React.FC<CreateSkillModalProps> = (props) => {
       manifestUrl: currentManifestUrl,
     });
     setSkillManifest(null);
+    setShowDetail(false);
   };
 
   const validateUrl = useCallback(
     (event) => {
       event.preventDefault();
       setShowDetail(true);
-      getSkillManifest(projectId, formData.manifestUrl, setSkillManifest, setFormDataErrors);
+      const localManifestPath = formData.manifestUrl.replace(/\\/g, '/');
+      getSkillManifest(projectId, formData.manifestUrl, setSkillManifest, setFormDataErrors, setShowDetail);
+      setManifestDirPath(localManifestPath.substring(0, localManifestPath.lastIndexOf('/')));
     },
     [projectId, formData]
   );
 
+  const validateLocalZip = useCallback(async (files: Record<string, JSZipObject>) => {
+    try {
+      // get manifest
+      const manifestFiles: JSZipObject[] = [];
+      const zipContent: Record<string, string> = {};
+      for (const fPath in files) {
+        zipContent[fPath] = await files[fPath].async('string');
+        // eslint-disable-next-line no-useless-escape
+        if (fPath.match(/\.([^\.]+)$/)?.[1] === 'json' && isManifestJson(zipContent[fPath])) {
+          manifestFiles.push(files[fPath]);
+          setManifestDirPath(fPath.substr(0, fPath.lastIndexOf('/') + 1));
+        }
+      }
+
+      // update content for detail panel and show it
+      if (manifestFiles.length > 1) {
```

Comment:
![image](https://user-images.githubusercontent.com/32191031/123612712-60925080-d835-11eb-836f-0425d2e03c55.png)

Add a condition for no manifest.json?